src/glauth.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Provides glauth class to control glauth."""
+
+import logging
+import shlex
```

Comment:
I do not think you need to use `shlex` in this case since you executing relatively simple commands. See comment below for more info.

Reply:
I got in the habit of using shlex since in the subprocess docs they mention it for guaranteed command success but for something this simple yeah not necessary. Updated!